### DISCUSSION TITLE
Export / share timestamps from favorite tracks

### DISCUSSION THREAD

**Discussion by printmaps** [devemux86/cruiser#687] [Q&A] [answered]
For all tracks stored in the internal track database: Please export / share also the timestamps.

Currently all timestamps are missing (coord and elevation are available). See example for an entry:

```
      <trkpt lat="51.95433" lon="7.53542">
        <ele>121.9</ele>
      </trkpt>

```

Testet with Cruiser 5.1.

And something like this is the expectation:

```
      <trkpt lat="51.9543372" lon="7.5354227">
        <ele>121.9000015258789</ele>
        <time>2024-12-15T06:32:43Z</time>
        <hdop>3.8389999866485596</hdop>
      </trkpt>

```

BTW: The second entry (directly exported from current track) is more accurate (e.g. 7.5354227 versus 7.53542).

**Comment by devemux86** [comment]
@printmaps Thanks for the suggestion.

There is already an existing thread on this topic:
- https://github.com/devemux86/cruiser/discussions/659

**Comment by devemux86** [comment]
@printmaps 

About the precision of coordinates, please see here:
https://wiki.openstreetmap.org/wiki/Precision_of_coordinates

- 5 decimals: precision 1 m
- 6 decimals: precision 1 cm

These at the equator, as at other latitudes, are more accurate.

Web apps / routers use 5 decimals precision in their responses,
as they need to consider the best between precision vs storage.

**Comment by printmaps** [comment]
I would save the recorded values unchanged. This gives downstream analyses all possibilities.

If the data has to be shortened, the coordinates and heights should have an accuracy of at least 1 mm. The time stamps should be accurate to a hundredth of a second.

**Comment by printmaps** [comment]
I would like to use the internal track database to manage all my tracks. At the moment, the feature is still quite rudimentary and can't really be used for this purpose without the timestamps in the tracks. However, I hope that it will continue to develop into a first-class feature.

**Reply by devemux86** [reply]
@printmaps Thanks for the detailed explanation. :+1: 

We have to think the storage vs memory vs precision.
As explained above, all web routers use the same coordinate precision for the same reason.

Nevertheless, I will think how we can improve this. :slightly_smiling_face:

**Comment by devemux86** [comment]
@printmaps Export / share timestamps from favorite tracks will available as part of the new:
- https://github.com/devemux86/cruiser/discussions/706

**Comment by devemux86** [comment]
Export / share timestamps from favorite tracks is available in:
- https://github.com/devemux86/cruiser/discussions/734

**Comment by devemux86** [comment, marked as answer]
- Export / share timestamps from favorite tracks is available in https://github.com/devemux86/cruiser/discussions/757
